### PR TITLE
Feature/acas cmpdreg merge

### PR DESCRIPTION
**Pull request by brianbolt**

### CODE REVIEW

File: src/main/java/com/labsynch/labseer/db/migration/postgres/V2_0_1_4__set_defaults.java [hunk 1]
```diff
@@ -0,0 +1,497 @@
+package com.labsynch.labseer.db.migration.postgres;
```
Comment (bffrost) [inline src/main/java/com/labsynch/labseer/db/migration/postgres/V2_0_1_4__set_defaults.java]: @brianbolt I'd broadly like a tour of how you went about merging the flyway migrations, and what the expected migration process will be

File: pom.xml [hunk 2]
```diff
@@ -28,24 +27,29 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.]]>
...
+		<gvnix.version>1.5.0.RELEASE</gvnix.version>
+		<querydsl.version>3.4.3</querydsl.version>
 		<spring-security.version>3.2.5.RELEASE</spring-security.version>
 		<cdk.version>1.5.13</cdk.version>
+		<jchem.version>16.4.25.0</jchem.version>
+		<indigo.version>1.2.3.r0</indigo.version>
 		<git-sha-1>${buildNumber}</git-sha-1>
 		<BuildScmBranch>${scmBranch}</BuildScmBranch>
-	</properties>
+	<tomcat.version>7.0.63</tomcat.version>
```
Comment (bffrost) [inline pom.xml]: Nit: indent is funny here

File: src/main/java/com/labsynch/labseer/utils/PostSuccessfulAuthenticationHandler.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.ldap.userdetails.LdapUserDetails;
+import org.springframework.security.web.authentication.SavedRequestAwareAuthenticationSuccessHandler;
+import org.springframework.security.web.savedrequest.HttpSessionRequestCache;
+import org.springframework.security.web.savedrequest.SavedRequest;
+
+public class PostSuccessfulAuthenticationHandler extends  SavedRequestAwareAuthenticationSuccessHandler {
```
Comment (bffrost) [inline src/main/java/com/labsynch/labseer/utils/PostSuccessfulAuthenticationHandler.java]: Has anything changed about how ACAS Roo and CReg do authentication? Is this file still necessary, and does it need any changes?

File: Dockerfile [hunk 4]
```diff
@@ -3,10 +3,10 @@ WORKDIR /src
 ENV CATALINA_HOME /usr/local/tomcat
 ENV PATH $CATALINA_HOME/bin:$PATH
 ADD 	pom.xml /src/pom.xml
-RUN 	["mvn", "dependency:resolve"]
+# RUN 	["mvn", "dependency:resolve", "-P", "indigo"]
```
[the remark this answers is not in the document]
Comment (brianbolt) [reply Dockerfile]: Going to add this back in and commit

File: pom.xml (hunk 2 above)
Comment (brianbolt) [reply pom.xml]: fixing with a new commit

File: src/main/java/com/labsynch/labseer/domain/Project.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.roo.addon.javabean.RooJavaBean;
+import org.springframework.roo.addon.jpa.activerecord.RooJpaActiveRecord;
+import org.springframework.roo.addon.json.RooJson;
+import org.springframework.roo.addon.tostring.RooToString;
+
+@RooJavaBean
+@RooToString
+@RooJpaActiveRecord(finders = { "findProjectsByCodeEquals", "findProjectsByNameEquals" })
+@RooJson
+public class Project {
```
[the remark this answers is not in the document]
Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/domain/Project.java]: Will be removing in a commit shortly.  While removing project, I found places it was still being used such as in the search form in Creg.  I am modifying creg search to use the same mechanism of passing in allowed projects as part of the request so that roo can filter.

File: pom.xml [hunk 6]
```diff
@@ -576,12 +730,21 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.]]>
 					<complianceLevel>${java.version}</complianceLevel>
 					<source>${java.version}</source>
 					<target>${java.version}</target>
+					<!-- Aspects in src/main/java and src/main/aspects are added as default 
+						in the compile goal. Aspects in src/test/java and src/test/aspects are added 
+						as default in the test-compile goal. Aspects in src/main/java are added in 
+						the test-compile goal if weaveWithAspectsInMainSourceFolder is set to true -->
+					<weaveWithAspectsInMainSourceFolder>false</weaveWithAspectsInMainSourceFolder>
```
[the remark this answers is not in the document]
Comment (brianbolt) [reply pom.xml]: It's described in the documentation as "Flag to indicate if aspects in the the main source dirs should be a part of the compile process". I'm not sure what our purpose for it was in creg but it came in with that change. Going to leave it this way for now.

File: src/main/java/com/labsynch/labseer/utils/PostSuccessfulAuthenticationHandler.java (hunk 3 above)
Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/utils/PostSuccessfulAuthenticationHandler.java]: Nothing about ACAS ROO or creg has changed with regards to authentication, this file works and should be kept in place.

File: src/main/resources/db/migration/postgres/V1.0.0.0__base_acas_schema_tables.sql [hunk 7]
```diff
@@ -2188,7 +2188,6 @@ REVOKE ALL ON SCHEMA acas FROM PUBLIC;
 REVOKE ALL ON SCHEMA acas FROM acas;
 GRANT ALL ON SCHEMA acas TO acas;
 GRANT ALL ON SCHEMA acas TO postgres;
-GRANT USAGE ON SCHEMA acas TO seurat;
```
Comment (bffrost) [inline src/main/resources/db/migration/postgres/V1.0.0.0__base_acas_schema_tables.sql]: The ones below make sense as we are getting rid of the "compound" schema. This one seems unrelated, but I guess we don't need to build in native base support for a "seurat" user.
